src/transformers/onnx/config.py

```diff
@@ -178,6 +196,21 @@ def atol_for_validation(self) -> float:
         """
         return 1e-5
 
+    @property
+    def is_torch_support_available(self) -> bool:
```

Comment:
The ViT export with dynamic axes only works on `torch` v1.9 or the nightly release (i.e. the upcoming v1.12). In other words, it's not possible to export ViT with one batch size and then run inference with a different batch size in v1.10.x

I couldn't pin down exactly what the source of the problem was in `torch` v1.10.x (it seems like one of the ops in `torch.onnx` is buggy), but since v1.12 is about to be released my proposal is to assign a minimum `torch` version in `OnnxConfig`. By default this is v1.8, but can be overridden to accommodate these model-specific exceptions.

I'm not sure whether this approach of being selective about `torch` versions is allowed in `transformers` so please let me know what you think @sgugger @LysandreJik @michaelbenayoun :)

Reply:
Ouch, one problem with my proposal is that the unit tests for ViT will fail if the user / the CI has a version < 1.12 installed. This seems like a bad idea, so will try to come up with an alternative solution

Reply:
As discussed internally, we can keep the current proposal but skip the ViT test if the `torch` versions are incompatible. I've added a skip in the tests so now 

```
RUN_SLOW=1 pytest tests/onnx/test_onnx_v2.py -k "vit" -rp
```

will be skipped if the user doesn't have the latest torch version installed.

Reply:
For torch.fx we have a requirement on a specific torch version. If you have validated that it doesn't work with a specific torch version, I would see no problem in printing a warning mentioning exactly that. If it's going to fail, then raising an error is also fine.